CriticalMass/MapViewController.swift

```diff
@@ -280,6 +285,64 @@ extension MapViewController: MKMapViewDelegate {
         }
         return renderer
     }
+
+    // MARK: Menu interaction handling
+
+    func mapView(_: MKMapView, didSelect view: MKAnnotationView) {
+        if view.annotation is CriticalMassAnnotation {
+            let longTapGestureRecognizer = UILongPressGestureRecognizer(
+                target: self,
+                action: #selector(handleEventLongPress(_:))
+            )
+            view.addGestureRecognizer(longTapGestureRecognizer)
+        }
+    }
+
+    @objc private func handleEventLongPress(_: Any) {
+        let routeAction = UIAlertAction(title: L10n.menuRoute, style: .default) { _ in self.routeToEvent() }
+        let shareAction = UIAlertAction(title: L10n.menuShare, style: .default) { _ in self.shareEvent() }
+        AlertPresenter.shared.presentAlert(
+            title: L10n.menuTitle,
+            preferredStyle: .actionSheet,
+            actionData: [routeAction, shareAction],
+            isCancelable: true
+        )
+    }
+
+    private func routeToEvent() {
+        nextRideManager.nextRide.flatMap {
+            $0.openInMaps()
+        }
```

Comment:
Shouldn't we use forEach here since we don't want to create a new collection

Reply:
I think i don’t get what you mean. But Maybe the GitHub app placed the comment wrong.

Where do you want to use it?

Reply:
Ah I though it is a collection, so my comment doesn't make sense..

Why are you using flatMap here and not simple optional changing?

Reply:
good. call didn't really pay attention